app/src/main/java/com/exallium/mvvmcleanexample/app/users/UserEditView.kt

```diff
@@ -25,34 +26,46 @@ class UserEditView : AppCompatActivity() {
     @Inject
     lateinit var router: AppRouter
 
-    val compositeDisposable = CompositeDisposable()
+    private var binding by autoEcoBinding<UserEditViewBinding>(null)
+    private var disposable by ecoDisposable(null)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view)
+        binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view).also { binding ->
```

Comment:
it is, but it is defined as a nullable var: 

```
private var binding by autoEcoBinding<UserEditViewBinding>(null)
// without inference
private var binding: UserEditViewBinding? by autoEcoBinding<UserEditViewBinding>(null)
```
The reason for the also block means that it is safely cast to non null type.
The binding wont be written to by another thread of course, I just wish I could tell kotlin that so it just smart casts anyway :/